browser/brave_vpn/android/vpn_native_worker.cc

```diff
@@ -0,0 +1,158 @@
+/* Copyright (c) 2021 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/browser/brave_vpn/android/vpn_native_worker.h"
+
+#include "base/android/jni_android.h"
+#include "base/android/jni_array.h"
+#include "base/android/jni_string.h"
+#include "base/bind.h"
+#include "base/json/json_writer.h"
+#include "base/values.h"
+#include "brave/browser/brave_vpn/vpn_service_factory.h"
+#include "brave/build/android/jni_headers/VpnNativeWorker_jni.h"
+#include "brave/components/brave_vpn/browser/vpn_service.h"
+#include "chrome/browser/profiles/profile.h"
+#include "chrome/browser/profiles/profile_manager.h"
+
+namespace chrome {
+namespace android {
+
+VpnNativeWorker::VpnNativeWorker(JNIEnv* env,
+                                 const base::android::JavaRef<jobject>& obj)
+    : weak_java_vpn_native_worker_(env, obj),
+      vpn_service_(nullptr),
+      weak_factory_(this) {
+  Java_VpnNativeWorker_setNativePtr(env, obj, reinterpret_cast<intptr_t>(this));
+  vpn_service_ = VpnServiceFactory::GetForProfile(
+      ProfileManager::GetActiveUserProfile()->GetOriginalProfile());
+}
+
+VpnNativeWorker::~VpnNativeWorker() {}
+
+void VpnNativeWorker::Destroy(
+    JNIEnv* env,
+    const base::android::JavaParamRef<jobject>& jcaller) {
+  delete this;
+}
+
+void VpnNativeWorker::GetAllServerRegions(
+    JNIEnv* env,
+    const base::android::JavaParamRef<jobject>& jcaller) {
+  if (vpn_service_) {
+    vpn_service_->GetAllServerRegions(base::Bind(
```

Comment:
We should avoid `base::Bind()` now. It's removed since C92.
and we also deleted all.  - https://github.com/brave/brave-core/pull/8811